lib/compat/wordpress-5.9/class-wp-theme-json-5-9.php

```diff
@@ -1540,42 +1577,40 @@ public function get_svg_filters( $origins ) {
 	}
 
 	/**
-	 * Returns whether a presets should be overriden or not.
+	 * For metadata values that can either be booleans or paths to booleans, gets the value.
 	 *
-	 * @param array      $theme_json The theme.json like structure to inspect.
-	 * @param array      $path Path to inspect.
-	 * @param bool|array $override Data to compute whether to override the preset.
+	 * ```php
+	 * $data = array(
+	 *   'color' => array(
+	 *     'defaultPalette' => true
+	 *   )
+	 * );
+	 *
+	 * static::get_metadata_boolean( $data, false );
+	 * // => false
+	 *
+	 * static::get_metadata_boolean( $data, array( 'color', 'defaultPalette' ) );
+	 * // => true
+	 * ```
+	 *
+	 * @param array      $data The data to inspect.
+	 * @param bool|array $path Boolean or path to a boolean.
+	 * @param bool       $default Default value if the referenced path is missing.
 	 * @return boolean
 	 */
-	protected static function should_override_preset( $theme_json, $path, $override ) {
-		if ( is_bool( $override ) ) {
-			return $override;
+	protected static function get_metadata_boolean( $data, $path, $default = false ) {
+		if ( is_bool( $path ) ) {
+			return $path;
 		}
 
-		// The relationship between whether to override the defaults
-		// and whether the defaults are enabled is inverse:
-		//
-		// - If defaults are enabled  => theme presets should not be overriden
-		// - If defaults are disabled => theme presets should be overriden
-		//
-		// For example, a theme sets defaultPalette to false,
-		// making the default palette hidden from the user.
-		// In that case, we want all the theme presets to be present,
-		// so they should override the defaults.
-		if ( is_array( $override ) ) {
-			$value = _wp_array_get( $theme_json, array_merge( $path, $override ) );
-			if ( isset( $value ) ) {
-				return ! $value;
-			}
-
-			// Search the top-level key if none was found for this node.
-			$value = _wp_array_get( $theme_json, array_merge( array( 'settings' ), $override ) );
+		if ( is_array( $path ) ) {
+			$value = _wp_array_get( $data, $path );
 			if ( isset( $value ) ) {
-				return ! $value;
+				return $value;
```

Comment:
this does seem like a very nice improvement over `should_override_preset` because the conditional inversion is surprising.